pkg/snap/util/dqlite_test.go

```diff
@@ -84,3 +85,72 @@ Role: 0`,
 	})
 
 }
+
+func TestUpdateDqliteBindAddress(t *testing.T) {
+	s := &mock.Snap{
+		DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0`,
+		DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+	}
+
+	g := NewWithT(t)
+	findMatchingBindAddressMock := func(addr string) (string, error) {
+		return "10.10.10.10:19001", nil
+	}
+
+	// should fail if node already known
+	err := snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1", "127.0.0.1", findMatchingBindAddressMock)
+	g.Expect(err).To(MatchError("the joining node (127.0.0.1) is already known to dqlite"))
+
+	s = &mock.Snap{
+		DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0
+- Address: 10.10.10.10:19001
+  ID: 12345678987654321
+  Role: 0`,
+		DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+	}
+	// should not update bind address if cluster consists of multiple nodes
+	err = snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1:19001", "8.8.8.8", findMatchingBindAddressMock)
+	g.Expect(err).To(BeNil())
+	g.Expect(s.WriteDqliteUpdateYamlCalledWith).To(BeEmpty())
+
+	s = &mock.Snap{
+		DqliteClusterYaml: `
+- Address: 127.0.0.1:19001
+  ID: 1236189235178654365
+  Role: 0`,
+		DqliteInfoYaml: `
+Address: 127.0.0.1:19001
+ID: 1236189235178654365
+Role: 0`,
+	}
+	// Update cluster yaml asynchronously
+	// The mock snap implementation doesn't actually write the yaml but instead
+	// sets `WriteDqliteUpdateYamlCalledWith`.
+	// Thus, we need to manually change the config as otherwise `WaitForDqliteCluster`
+	// would wait forever.
+	go func() {
+		<-time.After(500 * time.Millisecond)
+		s.DqliteClusterYaml = `
+- Address: 10.10.10.10:19001
+  ID: 1236189235178654365
+  Role: 0`
+	}()
+
+	// should update Dqlite bind address if cluster consists of one node and binds to local address
+	err = snaputil.MaybeUpdateDqliteBindAddress(context.Background(), s, "127.0.0.1:19001", "10.10.10.10", findMatchingBindAddressMock)
+	g.Expect(err).To(BeNil())
+	expectedCalledWith := []string{"Address: '[10.10.10.10:19001]:19001'\n"}
```

Comment:
This address does not look right to me, it should probably be in the form `$ip:$port`

Are you sure you are calling `MaybeUpdateDqliteBindAddress` properly?

Reply:
Oh boy, I shouldn't commit stuff before the first coffee... Thanks for the catch.

I was calling the `MaybeUpdateDqliteBindAddress` correct, but my `findMatchingBindAddress` mock was returning an IP + Port combination instead of just the IP.